Add core parts list and improve mechdb viewer

Icosahunter opened the pull request:
Fixes #33

Icosahunter committed:
Add core parts list and improve mechdb viewer

Icosahunter committed:
Fix default type selection.

infinitymdm reviewed (changes requested):
* Tanks don't appear in the database search results (as far as I can tell)
* The logo font doesn't seem to be rendering (at least not when using localhost)

Icosahunter committed:
Fix mechtronica logo, tanks missing from mechdb viewer, a few other fixes/features for mechdb viewer.

Icosahunter commented:
Fixed tanks not showing up (I hadn't added them to the core part group in the actual parts database apparently). Also fixed the logo SVG (apparently you have to convert text to paths when using non-standard fonts for them to be "portable" as it were).

I also added a message box if your search returns nothing/there was a syntax error in JSONPath (I don't think there is a way to tell which one happened...) as well as replaced undefined values with "---" in the result table.
